crates/astria-grpc-mock/src/mounted_mock.rs

```diff
@@ -125,6 +125,12 @@ impl MountedMock {
             return MockResult::NoMatch;
         }
 
+        if let Some(delay) = self.inner.delay {
```

Comment:
This is not the right place to do this: 

1. It's conceptually wrong: why does the mock only match after some time? Why does the mock not match before `duration`, but then start matching after `duration`?
2. instead, we should use a `tokio::time::sleep` future that actually responds after some time has passed
3. this cannot be done in `MountedMock::match_and_respond`, because it will stall the entire mock set. See the definition of `MockServer::handle_request` which locks the `RwLock` for the entire duration for handling the request